## Problem #70: Totient Permutation
[Link to Problem](https://projecteuler.net/problem=70)

### Problem Description

Euler's totient function, $\phi(n)$ [sometimes called the phi function], is used to determine the number of positive numbers less than or equal to $n$ which are relatively prime to $n$. For example, as $1, 2, 4, 5, 7$, and $8$, are all less than nine and relatively prime to nine, $\phi(9)=6$.<br>The number $1$ is considered to be relatively prime to every positive number, so $\phi(1)=1$.

Interestingly, $\phi(87109)=79180$, and it can be seen that $87109$ is a permutation of $79180$.

Find the value of $n$, $1 \lt n \lt 10^7$, for which $\phi(n)$ is a permutation of $n$ and the ratio $n/\phi(n)$ produces a minimum.

### Approach

Same as in [Problem 69](69.ipynb).

In [9]:
limit = 10 ** 7

phi = [i for i in range(limit)]

for i in range(2, limit):
    if phi[i] == i:
        for j in range(i, limit, i):
            phi[j] -= phi[j] // i

n = 87109

for i in range(2, limit):
    if sorted(str(i)) != sorted(str(phi[i])):
        continue
    if i / phi[i] < n / phi[n]:
        n = i

print(f"n = {n}, phi(n) = {phi[n]}")

n = 8319823, phi(n) = 8313928


###### Result: **8319823** | Execution time: ~32s

### Complexity analysis

Time complexity: $O(L \log\log L)$

##### Tags: #euler-totient, #sieve, #math